src/imu/inertial_measurement.hpp

```diff
@@ -20,15 +29,79 @@ class InertialMeasurement final : public InertialMeasurementInterface {
    * @param  i2c_handler Unique pointer to I2C Handler defined by hal driver 
    * 
    */
-  explicit InertialMeasurement(std::unique_ptr<i2c::I2CInterface> i2c_handler) : InertialMeasurementInterface(std::move(i2c_handler), std::make_unique<imu::Mpu9255>(std::move(i2c_handler))) {}
+  explicit InertialMeasurement(std::shared_ptr<i2c::I2CInterface> i2c_handler) : InertialMeasurementInterface(i2c_handler, std::make_unique<imu::Mpu9255>(i2c_handler)) {}
+
+  /**
+   * @brief Used for Initialization of complete Inertial Measurement Unit
```

Comment:
Documentation of these methods lack an explanation under what circumstances which return value is expected. How can you test these methods without knowing what they return under which circumstances

Reply:
Add of doxygen docu of return value.